### PR TITLE
feat: Allow for custom types

### LINKED ISSUE #39: Support custom tags
I've arrived to this repo from: https://github.com/zeke/semantic-pull-requests

As we are using GHE maybe we cannot use an app, but action looks great.

But, we use some tags that are not the ones defined on this action: `Feature`, `Fix`, Tech`...

Is possible to add the ability to allow tag customization?

Captura de pantalla 2020-10-01 a las 10 24 22

### LINKED ISSUE #45: Upon failure, report available options
When people are learning about conventional commits, they might not know all of the available options.

What might be really useful is that if there's a failure to match a conventional commit, to some how surface the list that's being checked against.

Some random ideas:
1. Leave a comment with all descriptions?
2. Leave a comment with a link to the [conventional-commit-types](https://github.com/commitizen/conventional-commit-types/blob/d1fb9cc953b8f1e33e2dd63babdf6ae8ab1c8711/index.json) file

### PR DESCRIPTION
**Pull request by amannn**
Based on https://github.com/amannn/action-semantic-pull-request/pull/47 and https://github.com/amannn/action-semantic-pull-request/pull/52.

It seems there are some issues with running the latest changes from a branch within a PR workflow, therefore I opened another PR which contains the relevant changes from #47 and #52.

The new behaviour can for example be seen here: https://github.com/amannn/action-semantic-pull-request/runs/1273430601

Resolves #39 
Resolves #45
Closes #47
Closes #52

### CODE REVIEW

File: .github/workflows/lint-pr-title.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: 'Lint PR title'
 on:
-  pull_request_target:
+  pull_request:
```
Comment (amannn) [inline .github/workflows/lint-pr-title.yml:3]: I've changed this to run checks against this branch.

File: README.md [hunk 2]
```diff
@@ -39,6 +39,10 @@ jobs:
       - uses: amannn/action-semantic-pull-request@v1.2.0
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        # Optionally you can configure which types are allowed.
+        # Default: https://github.com/commitizen/conventional-commit-types
+        with:
+          types: fix, feat
```
Comment (amannn) [inline README.md:45]: This is a string, as Github Actions doesn't support array arguments currently.

File: action.yml [hunk 3]
```diff
@@ -7,3 +7,6 @@ runs:
 branding:
   icon: 'shield'
   color: 'green'
+inputs:
+  types:
```
Comment (amannn) [inline action.yml:11]: Conventional commits calls these "types", therefore I'd go for the same terminology.

File: src/validatePrTitle.js [hunk 4]
```diff
@@ -2,24 +2,37 @@ const conventionalCommitsConfig = require('conventional-changelog-conventionalco
...
 const parser = require('conventional-commits-parser').sync;
 
-module.exports = async function validatePrTitle(prTitle) {
+const defaultTypes = Object.keys(conventionalCommitTypes.types);
+
+module.exports = async function validatePrTitle(prTitle, types = defaultTypes) {
   const {parserOpts} = await conventionalCommitsConfig();
   const result = parser(prTitle, parserOpts);
 
+  function printAvailableTypes() {
```
Comment (amannn) [inline src/validatePrTitle.js:11]: Sneaked in another improvement here: https://github.com/amannn/action-semantic-pull-request/issues/45

**Review by alorma — APPROVED** [review]
Good work!

**Comment by alorma** [comment]
Hey @amannn Will you merge this?

**Comment by amannn** [comment]
@alorma Yep, I've just published [v2.1.0](https://github.com/amannn/action-semantic-pull-request/releases/tag/v2.1.0) which contains this feature. I've mentioned you in the changelog for your contribution.

**Pull request merged**